include/layers/EWLayer.hpp

```diff
@@ -29,21 +36,22 @@ T relu(const T& value) {
 }
 
 template <typename ValueType>
-class EWLayer : public Layer<ValueType> {
+class EWLayerimpl : public Layerimpl<ValueType> {
```

Comment:
```suggestion
class EWLayerImpl : public Layerimpl<ValueType> {
```
same for other `*Impl` layers